### PR TITLE
[CPU] Add an e2e test for i4 store emulation.

### LINKED ISSUE #15369: Support sub-byte emulation for vector.store
https://github.com/llvm/llvm-project/pull/70293 implements the emulation logic, and we need to test if that work with IREE. Below is one of examples that can't be compiled now. We should make sure that we are able to add the test to `tests/e2e/linalg/f32_to_i4.mlir`.

```mlir
func.func @f32_to_i4_1d() {
  %input = util.unfoldable_constant dense<[0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0]> : tensor<8xf32>
  %0 = tensor.empty() : tensor<8xi4>
  %res = linalg.generic {indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>], iterator_types = ["parallel"]}
    ins(%input : tensor<8xf32>) outs(%0 : tensor<8xi4>) {
  ^bb0(%in: f32, %out: i4):
    %2 = arith.fptoui %in : f32 to i32
    %3 = arith.trunci %2 : i32 to i4
    linalg.yield %3 : i4
  } -> tensor<8xi4>
  check.expect_eq_const(%res, dense<[0, 1, 2, 3, 4, 5, 6, 7]> : tensor<8xi4>) : tensor<8xi4>
  return
}
```

### PR DESCRIPTION
**Pull request by hanhanW**
Fixes https://github.com/openxla/iree/issues/15369

### CODE REVIEW

**Comment by hanhanW** [comment]
We probably have to disable the test on some platforms because currently we don't support vector.maskedstore emulation (https://github.com/openxla/iree/issues/15540).

**Review by MaheshRavishankar — APPROVED** [review]

File: build_tools/cmake/test_riscv.sh [hunk 1]
```diff
@@ -99,6 +99,8 @@ test_exclude_args+=(
   "iree/tests/e2e/matmul/e2e_matmul_direct_f32_small_llvm-cpu_local-task"
   "iree/tests/e2e/matmul/e2e_matmul_direct_f32_small_no_padding_llvm-cpu_local-task"
   "iree/tests/e2e/regression/check_regression_llvm-cpu_strided_slice.mlir"
+  # TODO(#15540): Enable the tests.
+  "iree/tests/e2e/linalg/check_llvm-cpu_local-task_f32_to_i4.mlir"
```
Comment (hanhanW) [inline build_tools/cmake/test_riscv.sh]: @pzread @ScottTodd How do I disable the compilation for risc-v targets?

Comment (ScottTodd) [reply build_tools/cmake/test_riscv.sh]: Try the `noriscv` tag/label, as in https://github.com/openxla/iree/blob/2f9a1e173237b0948ef4f2bd3e03c85856d1be27/tests/e2e/matmul/BUILD.bazel#L39-L42

That should filter at _configure time_ (prior to _build time_), while this change to `test_riscv.sh` just filters at _test time_
https://github.com/openxla/iree/blob/fc441856b26be7691ed9406a7c8617befd07b036/build_tools/cmake/iree_check_test.cmake#L9-L11 https://github.com/openxla/iree/blob/fc441856b26be7691ed9406a7c8617befd07b036/build_tools/cmake/iree_check_test.cmake#L68-L71

Comment (hanhanW) [reply build_tools/cmake/test_riscv.sh]: oh cool, I did not know the tag. Let me give it a shot. Thanks!!

Comment (ScottTodd) [reply build_tools/cmake/test_riscv.sh]: btw: bazel_to_cmake should correctly propagate the BUILD file tag to the CMake file (your commit just modified the Bazel file)

Comment (hanhanW) [reply build_tools/cmake/test_riscv.sh]: ooops.. good catch. I forgot to run the script before commit.

Comment (hanhanW) [reply build_tools/cmake/test_riscv.sh]: It is still compiled for risc-v targets. Any thoughts?

Comment (ScottTodd) [reply build_tools/cmake/test_riscv.sh]: The failing compile is `check_winograd_llvm-cpu_local-task_f32_to_i4.mlir_module.vmfb` ([logs here](https://github.com/openxla/iree/actions/runs/6830324343/job/18578279271?pr=15539#step:7:1028))

The `check_winograd_llvm-cpu_local-task` suite is missing the label.

However, looking closer... I think you only actually want to skip the new `f32_to_i4` tests, not all of the tests in `LLVM_SRCS`. You could add a separate group like `LLVM_I4_SRCS` and add new test suites using those sources, then only apply the label to those suites.

Comment (hanhanW) [reply build_tools/cmake/test_riscv.sh]: okay... the winograd test suite should only contain `conv2d.mlir`. I believe that the preprocessing pass is only applied to the test. I can fix it...

Create a test suite for i4_store for now.

**Pull request merged**